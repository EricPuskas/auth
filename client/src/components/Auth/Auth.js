import React from "react";
// Components
import Header from "../Header/Header";
import Content from "../Content/Content";
import Footer from "../Footer/Footer";
import LoginForm from "../LoginForm/LoginForm";
import Logo from "../Logo/Logo";

const Auth = () => {
  return (
    <>
      <Logo />
      <Header route="login" />
      <Content />
      <LoginForm />
      <Footer />
    </>
  );
};

export default Auth;
