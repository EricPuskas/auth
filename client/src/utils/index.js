import checkPropTypes from "check-prop-types";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const isEmpty = value =>
  value === undefined ||
  value === null ||
  (typeof value === "object" && Object.keys(value).length === 0) ||
  (typeof value === "string" && value.trim().length === 0);

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const checkProps = (component, expectedProps) => {
  return checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );
};

export const isSecure = password => {
  let error_message = "";
  let lowerRegex = /[a-z]/;
  let upperRegex = /[A-Z]/;
  let numberRegex = /[0-9]/;
  let specialRegex = /[$@$!%*?&]/;

  if (isEmpty(password)) {
    error_message = "Password field is required.";
  }
  if (!password.match(lowerRegex))
    error_message = "Password must contain at least one lowercase letter.";
  if (!password.match(upperRegex))
    error_message = "Password must contain at least one uppercase letter.";
  if (!password.match(numberRegex))
    error_message = "Password must contain at least one number.";
  if (!password.match(specialRegex))
    error_message = "Password must contain at least a special character";
  if (password.length < 8 || password.length > 30) {
    error_message = "Password must be between 8 and 30 characters.";
  }
  return error_message;
};

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
