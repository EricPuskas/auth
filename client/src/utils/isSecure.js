import isEmpty from "./isEmpty";

const isSecure = password => {
  let error_message = "";
  let lowerRegex = /[a-z]/;
  let upperRegex = /[A-Z]/;
  let numberRegex = /[0-9]/;
  let specialRegex = /[$@$!%*?&]/;

  if (isEmpty(password)) {
    error_message = "Password field is required.";
  }
  if (!password.match(lowerRegex))
    error_message = "Password must contain at least one lowercase letter.";
  if (!password.match(upperRegex))
    error_message = "Password must contain at least one uppercase letter.";
  if (!password.match(numberRegex))
    error_message = "Password must contain at least one number.";
  if (!password.match(specialRegex))
    error_message = "Password must contain at least a special character";
  if (password.length < 8 || password.length > 30) {
    error_message = "Password must be between 8 and 30 characters.";
  }
  return error_message;
};

export default isSecure;
