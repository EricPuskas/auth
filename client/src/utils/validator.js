import { isSecure } from "./index";

const Validator = {
  login: data => {
    const errors = {};
    if (!data.password) {
      errors.password = "Password field is required.";
    } else {
      errors.password = isSecure(data.password);
    }
    if (!data.email) {
      errors.email = "Email field is required.";
    } else if (!/^[^@]+@[^@]+\.[^@]+$/.test(data.email)) {
      errors.email = "Please enter a valid email address (example@email.com).";
    }
    return errors;
  },
  register: data => {
    const errors = {};
    if (!data.password) {
      errors.password = "Password field is required.";
    } else {
      errors.password = isSecure(data.password);
    }
    if (!data.email) {
      errors.email = "Email field is required.";
    } else if (!/^[^@]+@[^@]+\.[^@]+$/.test(data.email)) {
      errors.email = "Please enter a valid email address (example@email.com).";
    }
    if (!data.firstName) {
      errors.firstName = "First Name field is required.";
    }
    if (!data.lastName) {
      errors.lastName = "Last Name field is required.";
    }
    return errors;
  }
};

export default Validator;
