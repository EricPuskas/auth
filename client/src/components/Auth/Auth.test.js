import React from "react";
import { shallow } from "enzyme";
import { testStore } from "../../utils";
import Auth from "./Auth";

const setup = (initState = {}) => {
  const store = testStore(initState);
  return shallow(<Auth store={store} />).debug();
};

describe("Auth Component", () => {
  const initState = {};
  let wrapper = setup(initState);
  it("Should pass snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
