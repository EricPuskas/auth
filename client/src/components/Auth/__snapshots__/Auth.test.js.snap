// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Component Should pass snapshot test 1`] = `
"<Fragment>
  <Logo />
  <Connect(Header) route=\\"login\\" />
  <Connect(Content) />
  <Connect(ReduxForm) />
  <Footer />
</Fragment>"
`;
